setup.rst

```diff
@@ -122,22 +123,161 @@ working only on pure Python code the pydebug build provides several useful
+``-Wno-unused-value -Wno-empty-body -Qunused-arguments``. You can set your
+``CFLAGS`` environment variable to these flags when running ``configure``.
+
+If you are using clang_ with ccache_, turn off the noisy
+``parentheses-equality`` warnings with the ``-Wno-parentheses-equality`` flag.
+These warnings are caused by clang not  having enough information to detect
+that extraneous parentheses in expanded macros are valid, because the
+preprocessing is done separately by ccache.
+
+If you are using LLVM 2.8, also use the ``-no-integrated-as`` flag in order to
+build the :py:mod:`ctypes` module (without the flag the rest of CPython will
+still build properly).
+
+
+.. _windows-compiling:
+
+Windows
+-------
+
+**Python 3.6** and later can use Microsoft Visual Studio 2017.  You can download
+and use any of the free or paid versions of `Visual Studio 2017`_.
+
+When installing Visual Studio 2017, select the **Python development** workload
+and the optional **Python native development tools** component to obtain all of
+the necessary build tools. If you do not already have git installed, you can
+find git for Windows on the **Individual components** tab of the installer.
+
+.. note:: If you want to build MSI installers, be aware that the build toolchain
+  for them has a dependency on the Microsoft .NET Framework Version 3.5 (which
+  may not be configured on recent versions of Windows, such as Windows 10). If
+  you are building on a recent Windows version, use the Control Panel (Programs
+  | Programs and Features | Turn Windows Features on or off) and ensure that the
+  entry ".NET Framework 3.5 (includes .NET 2.0 and 3.0)" is enabled.
+
+Your first build should use the command line to ensure any external dependencies
+are downloaded:
+
+.. code-block:: dosbatch
+
+   PCBuild\build.bat
+
+After this build succeeds, you can open the ``PCBuild\pcbuild.sln`` solution in
+Visual Studio to continue development.
+
+See the `readme`_ for more details on what other software is necessary and how
+to build.
+
+.. note:: **Python 2.7** uses Microsoft Visual Studio 2008, which is most easily
+   obtained through an MSDN subscription.  To use the build files in the
+   `PCbuild directory`_ you will also need Visual Studio 2010, see the `2.7
+   readme`_ for more details.  If you have VS 2008 but not 2010 you can use the
+   build files in the `PC/VS9.0 directory`_, see the `VS9 readme`_ for details.
+
+.. _Visual Studio 2017: https://www.visualstudio.com/
+.. _readme: https://github.com/python/cpython/blob/master/PCbuild/readme.txt
+.. _PCbuild directory: https://github.com/python/cpython/tree/2.7/PCbuild/
+.. _2.7 readme: https://github.com/python/cpython/blob/2.7/PCbuild/readme.txt
+.. _PC/VS9.0 directory: https://github.com/python/cpython/tree/2.7/PC/VS9.0/
+.. _VS9 readme: https://github.com/python/cpython/blob/2.7/PC/VS9.0/readme.txt
+
+
+.. _build-dependencies:
+
+Install dependencies
+====================
```

Comment:
This is now a new top-level section, that covers installing dependencies.
Since it also explains how to buiild them for macOS, perhaps the title should be changed to ``Install and build dependencies``